chapters/CRUD-Operations.md

```diff
@@ -207,612 +207,645 @@ Encode the **did:btcr2** identifier as follows:
+to btcr2.dev discussion forum
 
-It takes the following inputs:
+#### Algo 5. Deterministically Generate Initial DID Document {.tabbed .unnumbered}
 
-* `identifier` - a valid **did:btcr2** identifier; REQUIRED; string.
-* `identifierComponents` - The decoded parts of a **did:btcr2** identifier; REQUIRED; object.
-  * `idType` - the type of identifier (`KEY` or `EXTERNAL`); REQUIRED; string.
-  * `version` - the identifier version; REQUIRED; integer.
-  * `network` - the Bitcoin network used for the identifier; REQUIRED; string.
-  * `genesisBytes` - the originating public key; REQUIRED; bytes.
-* `resolutionOptions` - options that extends the default options per
-   the [DID Resolution specification](https://w3c.github.io/did-resolution/#did-resolution-options); OPTIONAL; object.
-  * `versionId` - the version of the DID and DID document, an incrementing integer starting
-    from 1; the below list is not intended to be exhaustive; OPTIONAL; integer.
-  * `versionTime` - a timestamp used during resolution as a bound for when to
-      stop resolving; OPTIONAL; integer.
-  * `sidecarData` - data necessary for resolving a DID such as ::BTCR2 Updates:: and
-    ::SMT:: proofs; OPTIONAL; object.
-  * `network` - the Bitcoin network used for resolution; OPTIONAL; string; default=`"bitcoin"`.
+This algorithm deterministically generates an ::Initial DID Document:: from a
+33-byte compressed representation of a secp256k1 public key encoded into the
+identifier as the ::Genesis Bytes::. The DID document generated must be a
+conformant DID document according to the DID core v1.1 specification.
 
-It returns the following output:
+The DID document must contain only the following properties:
 
-* `initialDocument` - the valid first version of a DID document for a given **did:btcr2** identifier.
+* A @context array containing the DID core v1.1 context url ("[https://www.w3.org/ns/did/v1.1](https://www.w3.org/ns/did/v1.1)" )
+  and the did:btcr2 context ("[https://btcr2.dev/context/v1](https://btc1.dev/context/v1)").
+* A verificationMethod array containing a single verificationMethod. This
+  verificationMethod must have an id value of "#initialKey" be of the type
+  Multikey with the publicKeyMultibase value a multikey encoding of the
+  secp256k1 public key bytes.
+* An array containing the verificationMethod id ‘#initialKey’ for each of the
+  verification relationships. assertionMethod, authentication,
+  capabilityInvocation, capabilityDelegation
+* A service array containing three BTCR2 Beacon services. The serviceEndpoint
+  for each of these services must be a BIP21 URI encoding of a bitcoin address
+  controlled by the public key. The three different address formats are. p2pkh,
+  p2wpkh, p2tr. The id value of the service must be "#initialP2PKH",
+  #initialP2WPKH” and "#initialP2TR" respectively. Finally, each of these
+  services must have a type set to the array \["BTCR2Beacon",
+  "SingletonBeacon"\]
```

Comment:
```suggestion
  services must have a type set to the string "SingletonBeacon"
```

We are not handling the additional beacon types in this PR. That is a discussion we want to have in person.

For now we just want to document what we have implemented in the code as much as possible